src/pieces/assets/assets_api.py

```diff
@@ -0,0 +1,235 @@
+				if item.deleted:
+					cls.assets_snapshot.pop(asset_id)
+				else:
+					cls.asset_queue.put(asset_id)  # Add asset_id to the queue
+					cls.asset_set.add(asset_id)  # Add asset_id to the set
+		cls.block = False # Remove the block to end the thread
+
+
+
+	@classmethod
+	def get_asset_snapshot(cls,asset_id:str):
+		asset = cls.assets_snapshot.get(asset_id)
+		if asset:
+			return asset
+		else:
+			return cls.update_asset_snapshot(asset_id)
+
+	
+
+	@classmethod
+	def edit_asset_name(cls,asset_id, new_name):
+		asset_api = AssetApi(Settings.api_client)
+
+		# Get the asset using the provided asset_id
+		asset = cls.get_asset_snapshot(asset_id)
+
+		# Check if the existing name is found and update it
+		existing_name = asset.name
+		asset.name = new_name
+		print(f"Asset name changed from '{existing_name}' to '{new_name}'")
+
+		# Update the asset using the API
+		try:
+			asset_api.asset_update(asset=asset, transferables=False)
+			print("Asset name updated successfully.")
+		except Exception as e:
+			show_error("Error updating asset: ",{e})
+
+
+	@staticmethod
+	def delete_asset_by_id(asset_id):
+		delete_instance = AssetsApi(Settings.api_client)
+
+		try:
+			response = delete_instance.assets_delete_asset(asset_id)
+			return response
+		except Exception as e:
+			return f"Failed to delete {asset_id}"
+
+	@staticmethod
+	def reclassify_asset(asset_id, classification):
+		asset_api = AssetApi(Settings.api_client)
+		with open(Settings.extensions_dir) as f:
+			extension_mapping = json.load(f)
+			if classification not in extension_mapping:
+				show_error(f"Invalid classification: {classification}","Please choose from the following: \n "+", ".join(extension_mapping.keys()))
+				return
+			
+		try:
+			asset = asset_api.asset_snapshot(asset_id)
+			if asset.original.reference.classification.generic == ClassificationGenericEnum.IMAGE:
+				show_error("Error in reclassify asset","Image original format is not supported")
+				return
+			asset_api.asset_reclassify(asset_reclassification=AssetReclassification(ext=classification,asset=asset),
+												transferables=False)
+			print(f"reclassify {asset.name} the asset to {classification} successfully")
+		except Exception as e:
+			show_error("Error reclassifying asset: ",{e})
+
+	@staticmethod
+	def update_asset_value(file_path,asset_id):
+		try:
+			with open(file_path,"r") as f:
+				data = f.read()
+		except FileNotFoundError:
+			show_error("Error in update asset","File not found")
+			return
+		asset_api = AssetApi(Settings.api_client)
+		format_api = FormatApi(Settings.api_client)
+
+		# get asset
+		created = asset_api.asset_snapshot(asset_id, transferables=False)
+
+		# update the original format's value
+		original = format_api.format_snapshot(created.original.id, transferable=True)
+		if original.classification.generic == ClassificationGenericEnum.IMAGE:
+			show_error("Error in update asset","Original format is not supported")
+			return
+		if original.fragment.string.raw:
```

Comment:
if original.fragment.string does not exist this will end up crashing the program. I'm not sure how we can do this in python but in general it is very important to have proper null safety throughout the application.

Reply:
Would you mind adding a github issue to that